pkg/app/piped/imagewatcher/watcher.go

```diff
@@ -96,90 +100,99 @@ func (w *watcher) run(ctx context.Context, provider imageprovider.Provider, inte
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			targets := w.collectTargets(ctx, provider)
-			outdated, err := determineUpdates(ctx, targets, provider)
-			if err != nil {
-				w.logger.Error("failed to determine which one should be updated", zap.Error(err))
-				continue
+			updates := make([]config.ImageWatcherTarget, 0)
+			for id, repo := range w.gitRepos {
+				u, err := w.determineUpdates(ctx, id, repo, provider)
+				if err != nil {
+					w.logger.Error("failed to determine images to be updated",
+						zap.String("repo-id", id),
+						zap.Error(err),
+					)
+					continue
+				}
+				updates = append(updates, u...)
 			}
-			if len(outdated) == 0 {
+			if len(updates) == 0 {
 				w.logger.Info("no image to be updated")
 				continue
 			}
-			if err := update(outdated); err != nil {
+			if err := update(updates); err != nil {
 				w.logger.Error("failed to update image", zap.Error(err))
 				continue
 			}
 		}
 	}
 }
 
-// collectTarget collects target images for each git repository.
-func (w *watcher) collectTargets(ctx context.Context, provider imageprovider.Provider) (targets []config.ImageWatcherTarget) {
-	for id, repo := range w.gitRepos {
-		branch := repo.GetClonedBranch()
-		w.mu.Lock()
-		err := repo.Pull(ctx, branch)
-		w.mu.Unlock()
-		if err != nil {
-			w.logger.Error("failed to update repository branch",
-				zap.String("repo-id", id),
-				zap.Error(err),
-			)
+// determineUpdates gives back target images to be updated for a given repo.
+func (w *watcher) determineUpdates(ctx context.Context, repoID string, repo git.Repo, provider imageprovider.Provider) ([]config.ImageWatcherTarget, error) {
+	branch := repo.GetClonedBranch()
+	w.mu.Lock()
+	err := repo.Pull(ctx, branch)
+	w.mu.Unlock()
+	if err != nil {
+		return nil, fmt.Errorf("failed to update repository branch: %w", err)
```

Comment:
nit: "failed to get update from branch" or "failed to pull branch"

Reply:
Yup, you're right, thank you!